webrender/src/device/gfx.rs

```diff
@@ -1170,17 +1171,56 @@ impl<B: hal::Backend> Program<B> {
 
         let (vs_module, fs_module) = shader_modules.get(shader_name).unwrap();
 
+        let mut constants = Vec::new();
+        for i in 0 .. 7 {
+            constants.push(hal::pso::SpecializationConstant {
+                id: i,
+                range: (4 * i) as _ .. (4 * (i + 1)) as _,
+            });
+        }
+
+        let mut data: Vec<u8> = vec![0; 28];
```

Comment:
Maybe you could add a comment what should this match? Or it should be modified together with base.glsl, or something like this.